service/worker/deployment/deployment_client.go

```diff
@@ -687,6 +687,10 @@ func (d *DeploymentClientImpl) updateWithStart(
 		// successful but un-completed responses.
 		res, err := d.historyClient.ExecuteMultiOperation(ctx, multiOpReq)
 		if err != nil {
+			var multiErr *serviceerror.MultiOperationExecution
+			if errors.As(err, &multiErr) {
```

Comment:
@stephanos are all multi-op errors retryable or we should filter out some?

Reply:
hmm... it seems this does not catch the multi operation errors because it does not check for wrapped errors :(
``` go
func IsServiceHandlerRetryableError(err error) bool {
	if err.Error() == ErrNamespaceHandover.Error() {
		return false
	}

	switch err.(type) {
	case *serviceerror.Internal,
		*serviceerror.Unavailable:
		return true
	}

	return false
}
```